### ISSUE TITLE
TypeCompiler will not escape a single quotation mark in a Type.Literal schema item

### ISSUE THREAD

**Issue by MrBazlow** [sinclairzx81/typebox#539]
When TypeCompiler is compiling a Type.Literal that is a string which contains a single quotation mark `'`, it will escape and throw an error. Love the library by the way, absolutely amazing work.

Example: [stack blitz demo](https://stackblitz.com/edit/node-yzey3e?file=index.ts)

I think it's cause by the [yield value of this iterator](https://github.com/sinclairzx81/typebox/blob/483a0f7093be6e970a1043653c1c819af02af31c/src/compiler/compiler.ts#L266C47-L266C47)

**Comment by sinclairzx81** [comment]
@MrBazlow Hi, thanks for reporting! This is a great catch!

Have just published a patch for this on `0.31.2`. Updated generated code as follows.

```typescript
import { TypeCompiler } from '@sinclair/typebox/compiler'
import { Type } from '@sinclair/typebox'

const C = TypeCompiler.Code(Type.Literal("it's"))

console.log(C)

// return function check(value) {
//   return (
//     (value === 'it\'s') <--- escaped
//   )
// }
```
Things should be good to go, but if you want to give things a quick test your side, can close off the issue.
Thanks again!
S

**Comment by MrBazlow** [comment]
Updated, tested and works great. Many thanks!
